test-data/unit/README.md

```diff
@@ -0,0 +1,92 @@
+Tests
+=====
+
+
+Unit tests
+----------
+
+To add a simple unit test for a new feature you developed, open or create a
+`test-data/unit/check-*.test` file with a name that roughly relates to the
+feature you added (or create a new file).
+
+Add the test in this format anywhere in the file:
+
+    [case testNewFeature]
+    # you expect these lines to pass type check
+    # note: it will NOT be executed, just type checked
+    ...
+    ...
+    ...
+    # you expect this line to fail with a message 'Error!'
+    # note a required space after E:
+    ...  # E: Error!
+    # more lines that pass
+    ...
+    ...
+    # more failing lines
+    ... #  E: Error!
+    ... #  E: Error!
+
+    # the optional [builtins ...] tells the type checker to use stubs from the
+    # indicated file (see Fixtures section for details)
+    [builtins fixtures/isinstancelist.pyi]
+
+    # the optional [out] indicates that any text after it contains the expected
```

Comment:
Should we say that "E: " is preferred because it makes it easier to associate the errors with the code generating them at a glance, and it makes it easier to change the code of the test without having to change line numbers in `[out]`?

Also would be helpful to add an example out line.